### PR TITLE
login route

### PR DESCRIPTION
**Pull request by MohammedIrheem**
**Login Route**
* Make validation for email and password using hai/joi
* Throw errors using Boom
* Create a query to check whether the email exists or not.
* Compare the inserted password with the stored one.
* Generate token when the comparison is true.
* Add the generated token to the cookies.
* Make test for a check email query.

### CODE REVIEW

File: server/__tests__/routes/users.test.js [hunk 1]
```diff
@@ -1,6 +1,65 @@
...
+    const reqBody = {
+      email: 'Conrad_Rosenbaum38@gmail.com',
+      password: '123456',
+    };
+    await request(app)
+      .post('/api/v1/login')
+      .send(reqBody)
+      .set('Accept', 'application/json')
+      .expect(200);
+  });
```
Comment (AhmedSafi97) [inline server/__tests__/routes/users.test.js:18]: Good test, but you need to test if the token is sent or not !

File: server/__tests__/routes/users.test.js [hunk 2]
```diff
@@ -1,6 +1,65 @@
...
+      .post('/api/v1/login')
+      .send(reqBody)
+      .set('Accept', 'application/json')
+      .expect(400);
+    expect(res.body.message).toStrictEqual(
+      'Your password has to be at least six characters or numbers'
+    );
+  });
+
+  it('return status 401 for Not existed email', async () => {
```
Comment (AhmedSafi97) [inline server/__tests__/routes/users.test.js:36]: if the email doesn't exist then it's a bad request ! status code should be 400

File: server/__tests__/routes/users.test.js [hunk 3]
```diff
@@ -1,6 +1,65 @@
...
+      .post('/api/v1/login')
+      .send(reqBody)
+      .set('Accept', 'application/json')
+      .expect(401);
+    expect(res.body.message).toStrictEqual(
+      'Email does not exist, signup first'
+    );
+  });
+
+  it('return status 401 for in valid password', async () => {
```
Comment (AhmedSafi97) [inline server/__tests__/routes/users.test.js:52]: if the password is invalid then it's a bad request, status code should be 400!

File: server/src/controller/auth.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      } else {
+        const token = jwt.sign({ id: dataValues.id }, process.env.SECRET_KEY);
+        res.cookie('token', token).status(200);
+        res.json({ msg: 'logged in successfully' });
+      }
+    } else {
+      throw Boom.badRequest('Email does not exist, signup first');
+    }
+  } catch (error) {
+    if (error) next(error);
```
Comment (Amoodaa) [inline server/src/controller/auth.js:35]:
```suggestion
    next(error);
```

**Pull request merged**